server/src/main/java/org/opensearch/indices/replication/common/ReplicationCollection.java

```diff
@@ -236,13 +237,13 @@ public boolean cancelForShard(ShardId shardId, String reason) {
     }
 
     /**
-     * check if a shard is currently replicating
+     * Get target for shard
      *
-     * @param shardId      shardId for which to check if replicating
-     * @return true if shard is currently replicating
+     * @param shardId      shardId
+     * @return Optional ReplicationTarget for input shardId
      */
-    public boolean isShardReplicating(ShardId shardId) {
-        return onGoingTargetEvents.values().stream().anyMatch(t -> t.indexShard.shardId().equals(shardId));
+    public Optional<T> getOngoingReplicationTarget(ShardId shardId) {
+        return onGoingTargetEvents.values().stream().filter(t -> t.indexShard.shardId().equals(shardId)).findFirst();
```

Comment:
Thinking we should add an assert here that there is only a single entry?

Reply:
I think we should also assert while adding starting new replications. Existing asserts is on target id which is AtomicLong and will always be unique. 
```
T existingTarget = onGoingTargetEvents.putIfAbsent(target.getId(), target);
assert existingTarget == null : "found two Target instances with the same id";
```

Reply:
I will take this in a follow up PR. Let me know if you think otherwise.